File: python/src/fhir_openehr_bridge/service.py
"""Translation façade mirroring the .NET ITranslationService."""

from __future__ import annotations

from typing import Any

from .mappers import composition_to_fhir_bundle, fhir_patient_to_composition
from .models import TranslationResult, ValidationIssue


class TranslationService:
    def fhir_to_openehr(self, fhir_json: str) -> TranslationResult:
        return fhir_patient_to_composition(fhir_json)

    def openehr_to_fhir(self, composition: dict[str, Any] | None) -> TranslationResult:
        if composition is None:
            return TranslationResult(False, None, [ValidationIssue("error", "openEHR composition is null.")])
        return composition_to_fhir_bundle(composition)

File: rust/src/service.rs
//! Translation façade mirroring the .NET ITranslationService.

use crate::mappers::{composition_to_fhir_bundle, fhir_patient_to_composition};
use crate::models::{OpenEhrComposition, Outcome};
use serde_json::Value;

#[derive(Debug, Default, Clone, Copy)]
pub struct TranslationService;

impl TranslationService {
    pub fn new() -> Self {
        Self
    }

    pub fn fhir_to_openehr(&self, fhir_json: &str) -> Outcome<OpenEhrComposition> {
        fhir_patient_to_composition(fhir_json)
    }

    pub fn openehr_to_fhir(&self, composition: &OpenEhrComposition) -> Outcome<Value> {
        composition_to_fhir_bundle(composition)
    }
}
